util/git/client.go

```diff
@@ -623,7 +631,7 @@ func (m *nativeGitClient) runCredentialedCmd(command string, args ...string) err
 
 func (m *nativeGitClient) runCmdOutput(cmd *exec.Cmd) (string, error) {
 	cmd.Dir = m.root
-	cmd.Env = append(cmd.Env, os.Environ()...)
+	cmd.Env = append(os.Environ(), cmd.Env...)
```

Comment:
I thought about it but this looks like a pretty dangerous change to me. Currently, there is no known harm in exposing all env variables to `git` so I propose to keep it as is. The order of env variables had been changed here so that I could "override"  system env variable.